### PR TITLE
tvm: init at 0.7.0

### PR DESCRIPTION
**Pull request by adelbertc**
###### Motivation for this change
New package: [TVM](https://tvm.apache.org/), An End to End Deep Learning Compiler Stack for CPUs, GPUs and accelerators.

cc @imalsogreg @tkonolige 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux) <-- does this work on Darwin ATM? It didn't last time I checked.
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`) <-- None, shared headers and library only
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Result of a Nix build

```shell
$ ls -R result
result:
include  lib

result/include:
tvm

result/include/tvm:
runtime

result/include/tvm/runtime:
c_backend_api.h  container.h  data_type.h   memory.h  module.h	 object.h	registry.h    threading_backend.h
c_runtime_api.h  crt	      device_api.h  micro     ndarray.h  packed_func.h	serializer.h  vm

result/include/tvm/runtime/crt:
crt.h  error_codes.h  func_registry.h  graph_runtime.h	logging.h  memory.h  module.h  packed_func.h  platform.h  rpc_common  utvm_rpc_server.h

result/include/tvm/runtime/crt/rpc_common:
frame_buffer.h	framing.h  session.h  write_stream.h

result/include/tvm/runtime/micro:
standalone

result/include/tvm/runtime/micro/standalone:
utvm_runtime.h

result/include/tvm/runtime/vm:
bytecode.h  executable.h  memory_manager.h  vm.h

result/lib:
libtvm.dylib  libtvm_runtime.dylib
```

### CODE REVIEW

File: pkgs/development/compilers/tvm/default.nix
Comment (r-burns) [inline pkgs/development/compilers/tvm/default.nix:19]: If this works on all platforms, be sure to add an explicit `platforms = platforms.all;` so it is built by hydra.

File: pkgs/development/compilers/tvm/default.nix
Comment (r-burns) [inline pkgs/development/compilers/tvm/default.nix:17]: AFAIR parallel building is implied when using cmake

**Review by r-burns — CHANGES_REQUESTED** [review]
Builds on darwin. Some quick suggestions but otherwise LGTM. Oh, and remember to fixup your commits before merging.

File: pkgs/development/compilers/tvm/default.nix
Comment (SuperSandro2000) [reply pkgs/development/compilers/tvm/default.nix:19]: Unrelated to this PR: I have set this nowhere for my packages and they are all build on hydra for darwin and NixOS if I did not misinterpret that. To be honest I don't understand that flag fully.

**Comment by adelbertc** [comment]
@r-burns Feedback addresses, commits fixed up :-)

**Comment by imalsogreg** [comment]
This looks good to me! Builds fine on my linux box.

File: pkgs/development/compilers/tvm/default.nix
Comment (sikmir) [inline pkgs/development/compilers/tvm/default.nix:5]: Use `fetchFromGitHub` instead.

File: pkgs/development/compilers/tvm/default.nix
Comment (sikmir) [inline pkgs/development/compilers/tvm/default.nix:10]: It looks like `git` is not needed.

File: pkgs/development/compilers/tvm/default.nix
Comment (adelbertc) [reply pkgs/development/compilers/tvm/default.nix:5]: Happy to make the change, may I ask the reasoning? Doesn't `fetchFromGitHub` delegate to fetchgit`, the primary differences being ability to clone private repos (not relevant in this case) and not doing a submodule clone by default?

File: pkgs/development/compilers/tvm/default.nix
Comment (adelbertc) [reply pkgs/development/compilers/tvm/default.nix:10]: The build does have some logic for detecting if it is a git checkout, but the build can succeed without it with a warning I believe.

File: pkgs/development/compilers/tvm/default.nix
Comment (sikmir) [reply pkgs/development/compilers/tvm/default.nix:5]: `fetchFromGitHub` uses `fetchzip` under the hood in such case, that's much faster.
https://github.com/NixOS/nixpkgs/blob/f2d8aebe8e597a2ba9b13769fbb0fcaa06b14b16/pkgs/build-support/fetchgithub/default.nix#L14

Comment (adelbertc) [reply pkgs/development/compilers/tvm/default.nix:5]: Ah in this case we do need to `fetchSubmodules` so it would be identical to `fetchgit` right?

Comment (sikmir) [reply pkgs/development/compilers/tvm/default.nix:5]: Yes, but anyway `fetchFromGitHub` is more appropriate function for packages from GitHub.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101728` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tvm</li>
  </ul>
</details>

File: pkgs/development/compilers/tvm/default.nix
Comment (adelbertc) [reply pkgs/development/compilers/tvm/default.nix]: Fair enough, done!

File: pkgs/development/compilers/tvm/default.nix
Comment (sikmir) [reply pkgs/development/compilers/tvm/default.nix:13]: Yes, it tries to get comit hash to embed it into binary (https://github.com/apache/incubator-tvm/pull/6280). But even `git` command is available, it will not work, since by default `fetchgit` removes `.git/` folder after cloning repo.

Comment (adelbertc) [reply pkgs/development/compilers/tvm/default.nix]: Ahh I didn't know about the `.git/` bit, fixed!

**Comment by adelbertc** [comment]
@sikmir Are you able to merge this?

**Comment by sikmir** [comment]
> @sikmir Are you able to merge this?

I'm only maintainer, not a commiter.

File: pkgs/development/compilers/tvm/default.nix
Comment (SuperSandro2000) [inline pkgs/development/compilers/tvm/default.nix:6]:
```suggestion
  version = "0.7.0";

  src = fetchFromGitHub {
```

File: pkgs/development/compilers/tvm/default.nix
Comment (SuperSandro2000) [inline pkgs/development/compilers/tvm/default.nix:13]:
```suggestion
  };

  nativeBuildInputs = [ cmake ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101728` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tvm</li>
  </ul>
</details>

**Comment by adelbertc** [comment]
@SuperSandro2000 Good for merge? :D

**Comment by SuperSandro2000** [comment]
@adelbertc please squash the commit. After that it can be merged.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101728` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tvm</li>
  </ul>
</details>

**Comment by adelbertc** [comment]
@SuperSandro2000 done!

**Comment by adelbertc** [comment]
@SuperSandro2000 Any love? :D

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101728` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tvm</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101728` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tvm</li>
  </ul>
</details>

**Comment by adelbertc** [comment]
@SuperSandro2000 Thank you!! 🙏

**Pull request merged**
